PhpAmqpLib/Channel/AMQPChannel.php

```diff
@@ -844,13 +844,19 @@ public function basic_nack($delivery_tag, $multiple = false, $requeue = false)
      *
      * @param string $consumer_tag
      * @param bool $nowait
+     * @param bool $noreturn
      * @return mixed
      */
-    public function basic_cancel($consumer_tag, $nowait = false)
+    public function basic_cancel($consumer_tag, $nowait = false, $noreturn = false)
     {
         list($class_id, $method_id, $args) = $this->protocolWriter->basicCancel($consumer_tag, $nowait);
         $this->send_method_frame(array($class_id, $method_id), $args);
 
+        if ($nowait || $noreturn) {
```

Comment:
I still don't get why you need the `$noreturn`. Just set the `$nowait` var to `true` and it will do the same. What am I missing?